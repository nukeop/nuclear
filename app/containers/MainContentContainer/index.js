import React from 'react';
import { Route, Switch, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as Actions from '../../actions';

import MainLayout from '../../components/MainLayout';

import AlbumViewContainer from '../AlbumViewContainer';
import ArtistViewContainer from '../ArtistViewContainer';
import DashboardContainer from '../DashboardContainer';
import FavoritesContainer from '../FavoritesContainer';
import LyricsContainer from '../LyricsContainer';
import PlaylistsContainer from '../PlaylistsContainer';
import PlaylistViewContainer from '../PlaylistViewContainer';
import PluginsContainer from '../PluginsContainer';
import SearchResultsContainer from '../SearchResultsContainer';
import SettingsContainer from '../SettingsContainer';
import TagViewContainer from '../TagViewContainer';
import EqualizerViewContainer from '../EqualizerViewContainer';
import LibraryViewContainer from '../LibraryViewContainer';

import DownloadsContainer from '../../containers/DownloadsContainer';

class MainContentContainer extends React.Component {
  componentDidMount () {
    if (this.props.history &&
      this.props.location &&
      this.props.location.pathname === '/') {
      this.props.history.push('/dashboard');
    }
  }

  render () {
    return (
      <Route render={({ location }) => {
        return (
          <MainLayout>
            <Switch key={location.key} location={location}>
              <Route path='/album/:albumId' component={AlbumViewContainer} />
              <Route path='/artist/:artistId' component={ArtistViewContainer} />
              <Route path='/dashboard' component={DashboardContainer} />
              <Route path='/downloads' component={DownloadsContainer} />
              <Route path='/favorites/tracks' component={FavoritesContainer} />
              <Route path='/playlists' component={PlaylistsContainer} />
              <Route path='/playlist/:playlistId' component={PlaylistViewContainer} />
              <Route path='/plugins' component={PluginsContainer} />
              <Route path='/settings' component={SettingsContainer} />
              <Route path='/tag/:tagName' component={TagViewContainer} />
              <Route path='/search' component={SearchResultsContainer} />
              <Route path='/lyrics' component={LyricsContainer} />
              <Route path='/equalizer' component={EqualizerViewContainer} />
              <Route path='/library' component={LibraryViewContainer} />
            </Switch>
          </MainLayout>
        );
      }
      } />
    );
  }
}

function mapStateToProps () {
  return {};
}

function mapDispatchToProps (dispatch) {
  return {
    actions: bindActionCreators(Actions, dispatch)
  };
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(MainContentContainer));
