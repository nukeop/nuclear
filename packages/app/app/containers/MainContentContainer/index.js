import React from 'react';
import { Route, Switch, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as SearchActions from '../../actions/search';

import MainLayout from '../../components/MainLayout';

import AlbumViewContainer from '../AlbumViewContainer';
import ArtistViewContainer from '../ArtistViewContainer';
import DashboardContainer from '../DashboardContainer';
import DownloadsContainer from '../DownloadsContainer';
import EqualizerViewContainer from '../EqualizerViewContainer';
import FavoritesContainer from '../FavoritesContainer';
import LibraryViewContainer from '../LibraryViewContainer';
import LyricsContainer from '../LyricsContainer';
import PlaylistViewContainer from '../PlaylistViewContainer';
import PlaylistsContainer from '../PlaylistsContainer';
import PluginsContainer from '../PluginsContainer';
import SearchResultsContainer from '../SearchResultsContainer';
import SettingsContainer from '../SettingsContainer';
import TagViewContainer from '../TagViewContainer';
import VisualizerNode from '../VisualizerContainer/VisualizerNode';

class MainContentContainer extends React.Component {
  componentDidMount () {
    if (this.props.history &&
      this.props.location &&
      this.props.location.pathname === '/') {
      this.props.history.push('/dashboard');
    }
  }

  render () {
    return (
      <Route render={({ location }) => {
        return (
          <MainLayout>
            <Switch key={location.key} location={location}>
              <Route path='/album/:albumId' component={AlbumViewContainer} />
              <Route path='/artist/:artistId' component={ArtistViewContainer} />
              <Route path='/dashboard' component={DashboardContainer} />
              <Route path='/downloads' component={DownloadsContainer} />
              <Route path='/favorites/albums' component={FavoritesContainer} />
              <Route path='/favorites/tracks' component={FavoritesContainer} />
              <Route path='/favorites/artists' component={FavoritesContainer} />
              <Route path='/playlists' component={PlaylistsContainer} />
              <Route path='/playlist/:playlistId' component={PlaylistViewContainer} />
              <Route path='/plugins' component={PluginsContainer} />
              <Route path='/settings' component={SettingsContainer} />
              <Route path='/tag/:tagName' component={TagViewContainer} />
              <Route path='/search' component={SearchResultsContainer} />
              <Route path='/lyrics' component={LyricsContainer} />
              <Route path='/equalizer' component={EqualizerViewContainer} />
              <Route path='/visualizer' component={VisualizerNode} />
              <Route path='/library' component={LibraryViewContainer} />
            </Switch>
          </MainLayout>
        );
      }
      } />
    );
  }
}

function mapStateToProps () {
  return {};
}

function mapDispatchToProps (dispatch) {
  return {
    searchActions: bindActionCreators(SearchActions, dispatch)
  };
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(MainContentContainer));
