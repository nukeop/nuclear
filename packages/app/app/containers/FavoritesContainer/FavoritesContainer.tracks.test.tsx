
import React from 'react';
import { render, waitFor } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import _ from 'lodash';

import { buildStoreState } from '../../../test/storeBuilders';
import { AnyProps, configureMockStore, setupI18Next, TestRouterProvider, TestStoreProvider } from '../../../test/testUtils';
import MainContentContainer from '../MainContentContainer';

const updateStore = (key: string, value: object) => {
  // eslint-disable-next-line @typescript-eslint/no-var-requires
  const store = require('@nuclear/core').store;
  store.set(key, value);
};

describe('Track view container', () => {
  beforeAll(() => {
    setupI18Next();
  });

  beforeEach(() => {
    // eslint-disable-next-line @typescript-eslint/no-var-requires
    const { store } = require('@nuclear/core');
    store.clear();
  });

  it('should display favorite tracks', () => {
    const favorites = buildStoreState()
      .withFavorites()
      .build()
      .favorites;
    updateStore('favorites', favorites);
    const { component } = mountComponent();
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('should display empty state', () => {
    const { component } = mountComponent();
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('should show a popup when a track is clicked', async () => {
    const favorites = buildStoreState()
      .withFavorites()
      .build()
      .favorites;
    updateStore('favorites', favorites);
    const { component } = mountComponent();

    await waitFor(() => component.getAllByTestId('track-popup-trigger')[0].click());
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('should call provider.search when playing a track with no streams', async () => {
    const favorites = buildStoreState()
      .withFavorites()
      .build()
      .favorites;
    updateStore('favorites', favorites);

    const { component, store } = mountComponent();
    const state = store.getState();
    const selectedStreamProvider = _.find(
      state.plugin.plugins.streamProviders, 
      { sourceName: state.plugin.selected.streamProviders });

    await waitFor(() => component.getAllByTestId('play-now')[1].click());

    expect(selectedStreamProvider.search).toBeCalled();
    expect(selectedStreamProvider.getStreamForId).not.toBeCalled();
  });

  it('should call provider.search when playing a track from favorites', async () => {
    const favorites = buildStoreState()
      .withFavorites()
      .build()
      .favorites;
    updateStore('favorites', favorites);

    const { component, store } = mountComponent();
    const state = store.getState();
    const selectedStreamProvider = _.find(
      state.plugin.plugins.streamProviders, 
      { sourceName: state.plugin.selected.streamProviders });

    await waitFor(() => component.getAllByTestId('play-now')[0].click());

    expect(selectedStreamProvider.search).toHaveBeenCalledWith({
      artist: 'test artist 1',
      track: 'test track 1'
    });
  });

  it('should play a favorited local library track from a local stream', async () => {
    const favorites = buildStoreState()
      .withFavorites({
        tracks: [{
          uuid: 'local-track-1',
          artist: 'test artist 1',
          name: 'test track 1',
          local: true
        }]
      })
      .build()
      .favorites;
    updateStore('favorites', favorites);

    const { component, store } = mountComponent();
    await waitFor(() => component.getAllByTestId('play-now')[0].click());
    const state = store.getState();
    expect(state.queue.queueItems[0]).toEqual(expect.objectContaining({
      uuid: 'local-track-1',
      artist: 'test artist 1',
      name: 'test track 1',
      local: true,
      stream: {
        source: 'Local',
        stream: 'file:///home/nuclear/Music/local artist 1/local album 1/local track 1.mp3',
        duration: 300
      }
    }));
  });

  const mountComponent = (initialStore?: AnyProps) => {
    const initialState = initialStore ||
      buildStoreState()
        .withPlugins()
        .withConnectivity()
        .withPlaylists()
        .withLocal()
        .build();

    const history = createMemoryHistory({
      initialEntries: ['/favorites/tracks']
    });
    const store = configureMockStore(initialState);
    const component = render(
      <TestRouterProvider
        history={history}
      >
        <TestStoreProvider
          store={store}
        >
          <MainContentContainer />
        </TestStoreProvider>
      </TestRouterProvider >, {container: document.body}
    );
    return { component, history, store };
  };
});
