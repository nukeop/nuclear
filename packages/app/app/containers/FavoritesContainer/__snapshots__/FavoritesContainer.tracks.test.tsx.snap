// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Track view container should display empty state 1`] = `
<DocumentFragment>
  <div
    class="main_layout_container"
  >
    <div
      class="favorite_tracks_view"
    >
      <div
        class="empty_state"
      >
        <i
          class="icons"
        >
          <i
            aria-hidden="true"
            class="star icon"
          />
          <i
            aria-hidden="true"
            class="music corner icon"
          />
        </i>
        <h2>
          No favorites added
        </h2>
        <div>
          Try adding some tracks to favorites and they will appear here!
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Track view container should display favorite tracks 1`] = `
<DocumentFragment>
  <div
    class="main_layout_container"
  >
    <div
      class="favorite_tracks_view"
    >
      <div
        class="header_container"
      >
        Your favorite tracks
      </div>
      <div
        class="ui segment"
      >
        <table
          class="track_table"
          role="table"
        >
          <thead>
            <tr
              role="row"
            >
              <th
                colspan="1"
                role="columnheader"
              >
                Â 
              </th>
              <th
                colspan="1"
                role="columnheader"
              >
                <span
                  class="center_aligned"
                >
                  <i
                    aria-hidden="true"
                    class="hashtag icon"
                  />
                </span>
              </th>
              <th
                colspan="1"
                role="columnheader"
              >
                <span
                  class="center_aligned"
                >
                  <i
                    aria-hidden="true"
                    class="image icon"
                  />
                </span>
              </th>
              <th
                colspan="1"
                role="columnheader"
              >
                Title
              </th>
              <th
                colspan="1"
                role="columnheader"
              >
                Artist
              </th>
              <th
                colspan="1"
                role="columnheader"
              >
                <span
                  class="select_header"
                >
                  <div
                    class="ui fitted checkbox nuclear checkbox"
                    style="cursor: pointer;"
                  >
                    <input
                      class="hidden"
                      readonly=""
                      tabindex="0"
                      title="Toggle All Rows Selected"
                      type="checkbox"
                      value=""
                    />
                    <label />
                  </div>
                </span>
              </th>
            </tr>
          </thead>
          <tbody
            data-rbd-droppable-context-id="0"
            data-rbd-droppable-id="track_table"
            role="rowgroup"
          >
            <tr
              class=""
              data-rbd-draggable-context-id="0"
              data-rbd-draggable-id="test track 1 0"
              role="row"
            >
              <td
                class="narrow delete_cell"
                role="cell"
              >
                <button
                  class="ui tiny basic circular icon button nuclear button borderless"
                  data-testid="delete-button"
                >
                  <i
                    aria-hidden="true"
                    class="times icon"
                  />
                </button>
              </td>
              <td
                class="position_cell narrow"
                role="cell"
              >
                <button
                  class="ui pink tiny circular icon button nuclear button add_button"
                  data-testid="play-now"
                >
                  <i
                    aria-hidden="true"
                    class="play icon"
                  />
                </button>
                <span
                  class="position_cell_value"
                >
                  1
                </span>
              </td>
              <td
                class="thumbnail_cell narrow"
                role="cell"
              >
                <img
                  class="thumbnail"
                  src="https://test-track-thumb-url"
                />
              </td>
              <td
                class="title_cell"
                role="cell"
              >
                <span
                  class="title_cell_content"
                >
                  <span
                    class="title_cell_value"
                  >
                    test track 1
                  </span>
                  <span
                    class="title_cell_buttons"
                  >
                    <button
                      class="ui tiny basic circular icon button nuclear button borderless title_cell_button"
                      data-testid="add-to-queue"
                    >
                      <i
                        aria-hidden="true"
                        class="plus icon"
                      />
                    </button>
                    <button
                      class="ui tiny basic circular icon button nuclear button borderless title_cell_button"
                      data-testid="track-popup-trigger"
                    >
                      <i
                        aria-hidden="true"
                        class="ellipsis horizontal icon"
                      />
                    </button>
                  </span>
                </span>
              </td>
              <td
                role="cell"
              >
                test artist 1
              </td>
              <td
                class="select_cell narrow"
                role="cell"
              >
                <div
                  class="ui fitted checkbox nuclear checkbox"
                  style="cursor: pointer;"
                >
                  <input
                    class="hidden"
                    readonly=""
                    tabindex="0"
                    title="Toggle Row Selected"
                    type="checkbox"
                    value=""
                  />
                  <label />
                </div>
              </td>
            </tr>
            <tr
              class=""
              data-rbd-draggable-context-id="0"
              data-rbd-draggable-id="test track 2 1"
              role="row"
            >
              <td
                class="narrow delete_cell"
                role="cell"
              >
                <button
                  class="ui tiny basic circular icon button nuclear button borderless"
                  data-testid="delete-button"
                >
                  <i
                    aria-hidden="true"
                    class="times icon"
                  />
                </button>
              </td>
              <td
                class="position_cell narrow"
                role="cell"
              >
                <button
                  class="ui pink tiny circular icon button nuclear button add_button"
                  data-testid="play-now"
                >
                  <i
                    aria-hidden="true"
                    class="play icon"
                  />
                </button>
                <span
                  class="position_cell_value"
                >
                  2
                </span>
              </td>
              <td
                class="thumbnail_cell narrow"
                role="cell"
              >
                <img
                  class="thumbnail"
                  src="https://test-track-thumb-url"
                />
              </td>
              <td
                class="title_cell"
                role="cell"
              >
                <span
                  class="title_cell_content"
                >
                  <span
                    class="title_cell_value"
                  >
                    test track 2
                  </span>
                  <span
                    class="title_cell_buttons"
                  >
                    <button
                      class="ui tiny basic circular icon button nuclear button borderless title_cell_button"
                      data-testid="add-to-queue"
                    >
                      <i
                        aria-hidden="true"
                        class="plus icon"
                      />
                    </button>
                    <button
                      class="ui tiny basic circular icon button nuclear button borderless title_cell_button"
                      data-testid="track-popup-trigger"
                    >
                      <i
                        aria-hidden="true"
                        class="ellipsis horizontal icon"
                      />
                    </button>
                  </span>
                </span>
              </td>
              <td
                role="cell"
              >
                test artist 2
              </td>
              <td
                class="select_cell narrow"
                role="cell"
              >
                <div
                  class="ui fitted checkbox nuclear checkbox"
                  style="cursor: pointer;"
                >
                  <input
                    class="hidden"
                    readonly=""
                    tabindex="0"
                    title="Toggle Row Selected"
                    type="checkbox"
                    value=""
                  />
                  <label />
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
  <div
    aria-atomic="true"
    aria-live="assertive"
    id="rbd-announcement-0"
    style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
  />
  <div
    id="rbd-hidden-text-0-hidden-text-0"
    style="display: none;"
  >
    
  Press space bar to start a drag.
  When dragging you can use the arrow keys to move the item around and escape to cancel.
  Some screen readers may require you to be in focus mode or to use your pass through key

  </div>
</DocumentFragment>
`;

exports[`Track view container should show a popup when a track is clicked 1`] = `
<DocumentFragment>
  <div
    class="main_layout_container"
  >
    <div
      class="favorite_tracks_view"
    >
      <div
        class="header_container"
      >
        Your favorite tracks
      </div>
      <div
        class="ui segment"
      >
        <table
          class="track_table"
          role="table"
        >
          <thead>
            <tr
              role="row"
            >
              <th
                colspan="1"
                role="columnheader"
              >
                Â 
              </th>
              <th
                colspan="1"
                role="columnheader"
              >
                <span
                  class="center_aligned"
                >
                  <i
                    aria-hidden="true"
                    class="hashtag icon"
                  />
                </span>
              </th>
              <th
                colspan="1"
                role="columnheader"
              >
                <span
                  class="center_aligned"
                >
                  <i
                    aria-hidden="true"
                    class="image icon"
                  />
                </span>
              </th>
              <th
                colspan="1"
                role="columnheader"
              >
                Title
              </th>
              <th
                colspan="1"
                role="columnheader"
              >
                Artist
              </th>
              <th
                colspan="1"
                role="columnheader"
              >
                <span
                  class="select_header"
                >
                  <div
                    class="ui fitted checkbox nuclear checkbox"
                    style="cursor: pointer;"
                  >
                    <input
                      class="hidden"
                      readonly=""
                      tabindex="0"
                      title="Toggle All Rows Selected"
                      type="checkbox"
                      value=""
                    />
                    <label />
                  </div>
                </span>
              </th>
            </tr>
          </thead>
          <tbody
            data-rbd-droppable-context-id="1"
            data-rbd-droppable-id="track_table"
            role="rowgroup"
          >
            <tr
              class=""
              data-rbd-draggable-context-id="1"
              data-rbd-draggable-id="test track 1 0"
              role="row"
            >
              <td
                class="narrow delete_cell"
                role="cell"
              >
                <button
                  class="ui tiny basic circular icon button nuclear button borderless"
                  data-testid="delete-button"
                >
                  <i
                    aria-hidden="true"
                    class="times icon"
                  />
                </button>
              </td>
              <td
                class="position_cell narrow"
                role="cell"
              >
                <button
                  class="ui pink tiny circular icon button nuclear button add_button"
                  data-testid="play-now"
                >
                  <i
                    aria-hidden="true"
                    class="play icon"
                  />
                </button>
                <span
                  class="position_cell_value"
                >
                  1
                </span>
              </td>
              <td
                class="thumbnail_cell narrow"
                role="cell"
              >
                <img
                  class="thumbnail"
                  src="https://test-track-thumb-url"
                />
              </td>
              <td
                class="title_cell"
                role="cell"
              >
                <span
                  class="title_cell_content"
                >
                  <span
                    class="title_cell_value"
                  >
                    test track 1
                  </span>
                  <span
                    class="title_cell_buttons"
                  >
                    <button
                      class="ui tiny basic circular icon button nuclear button borderless title_cell_button"
                      data-testid="add-to-queue"
                    >
                      <i
                        aria-hidden="true"
                        class="plus icon"
                      />
                    </button>
                    <button
                      class="ui tiny basic circular icon button nuclear button borderless title_cell_button"
                      data-testid="track-popup-trigger"
                    >
                      <i
                        aria-hidden="true"
                        class="ellipsis horizontal icon"
                      />
                    </button>
                  </span>
                </span>
              </td>
              <td
                role="cell"
              >
                test artist 1
              </td>
              <td
                class="select_cell narrow"
                role="cell"
              >
                <div
                  class="ui fitted checkbox nuclear checkbox"
                  style="cursor: pointer;"
                >
                  <input
                    class="hidden"
                    readonly=""
                    tabindex="0"
                    title="Toggle Row Selected"
                    type="checkbox"
                    value=""
                  />
                  <label />
                </div>
              </td>
            </tr>
            <tr
              class=""
              data-rbd-draggable-context-id="1"
              data-rbd-draggable-id="test track 2 1"
              role="row"
            >
              <td
                class="narrow delete_cell"
                role="cell"
              >
                <button
                  class="ui tiny basic circular icon button nuclear button borderless"
                  data-testid="delete-button"
                >
                  <i
                    aria-hidden="true"
                    class="times icon"
                  />
                </button>
              </td>
              <td
                class="position_cell narrow"
                role="cell"
              >
                <button
                  class="ui pink tiny circular icon button nuclear button add_button"
                  data-testid="play-now"
                >
                  <i
                    aria-hidden="true"
                    class="play icon"
                  />
                </button>
                <span
                  class="position_cell_value"
                >
                  2
                </span>
              </td>
              <td
                class="thumbnail_cell narrow"
                role="cell"
              >
                <img
                  class="thumbnail"
                  src="https://test-track-thumb-url"
                />
              </td>
              <td
                class="title_cell"
                role="cell"
              >
                <span
                  class="title_cell_content"
                >
                  <span
                    class="title_cell_value"
                  >
                    test track 2
                  </span>
                  <span
                    class="title_cell_buttons"
                  >
                    <button
                      class="ui tiny basic circular icon button nuclear button borderless title_cell_button"
                      data-testid="add-to-queue"
                    >
                      <i
                        aria-hidden="true"
                        class="plus icon"
                      />
                    </button>
                    <button
                      class="ui tiny basic circular icon button nuclear button borderless title_cell_button"
                      data-testid="track-popup-trigger"
                    >
                      <i
                        aria-hidden="true"
                        class="ellipsis horizontal icon"
                      />
                    </button>
                  </span>
                </span>
              </td>
              <td
                role="cell"
              >
                test artist 2
              </td>
              <td
                class="select_cell narrow"
                role="cell"
              >
                <div
                  class="ui fitted checkbox nuclear checkbox"
                  style="cursor: pointer;"
                >
                  <input
                    class="hidden"
                    readonly=""
                    tabindex="0"
                    title="Toggle Row Selected"
                    type="checkbox"
                    value=""
                  />
                  <label />
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
  <div
    aria-atomic="true"
    aria-live="assertive"
    id="rbd-announcement-1"
    style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
  />
  <div
    id="rbd-hidden-text-1-hidden-text-4"
    style="display: none;"
  >
    
  Press space bar to start a drag.
  When dragging you can use the arrow keys to move the item around and escape to cancel.
  Some screen readers may require you to be in focus mode or to use your pass through key

  </div>
  <div
    style="display: flex; position: absolute; left: 0px; top: 0px; bottom: 0px; transform: translate(0px, 0px);"
  >
    <div
      class="ui top left popup transition visible context_popup"
      style="position: initial;"
    >
      <div
        class="popup_header"
      >
        <div
          class="popup_thumb"
        >
          <img
            src="https://test-track-thumb-url"
          />
        </div>
        <div
          class="popup_info"
        >
          <div
            class="popup_title"
          >
            test track 1
          </div>
          <div
            class="popup_artist"
          >
            by test artist 1
          </div>
        </div>
      </div>
      <hr />
      <div
        class="popup_buttons"
      >
        <a
          aria-label="Add track to queue"
          class="popup_button"
          data-testid="track-popup-add-queue"
          href="#"
        >
          <i
            aria-hidden="true"
            class="plus icon"
          />
           Add to queue
        </a>
        <a
          aria-label="Add track to play next"
          class="popup_button"
          href="#"
        >
          <i
            aria-hidden="true"
            class="play icon"
          />
           Play next
        </a>
        <a
          aria-label="Play this track now"
          class="popup_button"
          data-testid="track-popup-play-now"
          href="#"
        >
          <i
            aria-hidden="true"
            class="play icon"
          />
           Play now
        </a>
        <a
          aria-label="Add this track to favorites"
          class="popup_button"
          href="#"
        >
          <i
            aria-hidden="true"
            class="heart icon"
          />
           Add to favorites
        </a>
        <div
          aria-expanded="false"
          class="ui dropdown nuclear dropdown popup_button left"
          role="listbox"
          tabindex="0"
        >
          <div
            aria-atomic="true"
            aria-live="polite"
            class="divider text"
            role="alert"
          >
            Add to playlist
          </div>
          <i
            aria-hidden="true"
            class="dropdown icon"
          />
          <div
            class="menu transition"
          >
            <div
              class="item"
              role="option"
            >
              <i
                aria-hidden="true"
                class="music icon"
              />
              test playlist
            </div>
            <div
              class="item"
              role="option"
            >
              <i
                aria-hidden="true"
                class="music icon"
              />
              test playlist 2
            </div>
          </div>
        </div>
        <a
          aria-label="Download this track"
          class="popup_button"
          href="#"
        >
          <i
            aria-hidden="true"
            class="download icon"
          />
           Download
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
